components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -503,6 +503,58 @@ pub fn truncate_real(x: Real, d: i32) -> Real {
     }
 }
 
+const U64_TEN_POWS: [u64; 20] = [
+    1,
+    10,
+    100,
+    1_000,
+    10_000,
+    100_000,
+    1_000_000,
+    10_000_000,
+    100_000_000,
+    1_000_000_000,
+    10_000_000_000,
+    100_000_000_000,
+    1_000_000_000_000,
+    10_000_000_000_000,
+    100_000_000_000_000,
+    1_000_000_000_000_000,
+    10_000_000_000_000_000,
+    100_000_000_000_000_000,
+    1_000_000_000_000_000_000,
+    10_000_000_000_000_000_000,
+];
+
+#[inline]
+#[rpn_fn]
+pub fn truncate_uint_with_uint(arg0: &Option<Int>, arg1: &Option<Int>) -> Result<Option<Int>> {
```

Comment:
Could you tell why we need to treat it specially? It seems that TiDB always treat it as a signed int: https://github.com/pingcap/tidb/blob/master/expression/builtin_math.go#L1881

Reply:
@sticnarf As the [discussion](https://github.com/tikv/tikv/pull/3532#issuecomment-418701722) in #3532, if `arg1` is a very big u64, casting it to i64 will get a negative int.

For example, `truncate(1234, 18446744073709551615_u64)` should be 1234.
However, because `18446744073709551615_u64 as int64` is -1, `truncate(1234, 18446744073709551615_u64)` in TiDB will get 1230.